trieste/bayesian_optimizer.py

```diff
@@ -182,7 +183,20 @@ def optimize(
         num_steps: int,
         datasets: Mapping[str, Dataset],
         model_specs: Mapping[str, ModelSpec],
-        acquisition_rule: AcquisitionRule[S, SP],
+        acquisition_rule: AcquisitionRule[TensorType, SP],
+        *,
+        track_state: bool = True,
+        fit_intial_model: bool = True,
+    ) -> OptimizationResult[None]:
+        ...
+
+    @overload
+    def optimize(
+        self,
+        num_steps: int,
+        datasets: Mapping[str, Dataset],
+        model_specs: Mapping[str, ModelSpec],
+        acquisition_rule: AcquisitionRule[State[S | None, TensorType], SP],
```

Comment:
this is a slightly odd way to do this, allowing `None` as the return type, but it works and it means fewer changes